src/model_tool.py

```diff
@@ -58,9 +58,10 @@ def __init__(self,
         self.model_name = model_name
         self.model = None
         self.tokenizer = None
-        self.hparams = DEFAULT_HPARAMS
-        if hparams:
-            self.update_hparams(hparams)
+        self.hparams = {}
+        if hparams is None:
+            hparams = DEFAULT_HPARAMS
+        self.update_hparams(hparams)
```

Comment:
I think the above dosn't work if you just want to pass a partial hparams dictionary.  Could we change from:

if hparams is None:
        hparams = DEFAULT_HPARAMS

to:

self.update_hparams(DEFAULT_HPARAMS)


That should work right? Then a partial list should be good without having the same bug as before?

Reply:
Good catch. PTAL!